storage/shard.go

```diff
@@ -330,6 +331,9 @@ func (s *dbShard) tickAndExpire(
 	}
 	start := s.nowFn()
 	s.forEachShardEntry(func(entry *dbShardEntry) bool {
+		if c.IsCancelled() {
+			return false
+		}
```

Comment:
Perhaps move this into the `tickSleepIfAheadEvery` inner loop, so we only pause to check every 128 (which is still pretty frequent)?